solidity/src/Polymer7683.sol

```diff
@@ -0,0 +1,176 @@
+    error InvalidProof();
+    error InvalidChainId();
+    error InvalidEmitter();
+    error InvalidEventData();
+    error InvalidDestinationContract();
+    error UnregisteredDestinationChain();
+    error SettlementFailed();
+
+    // ============ Constructor ============
+    /**
+     * @notice Initializes the Polymer7683 contract with the specified Prover and PERMIT2 address.
+     * @param _prover The address of the Polymer CrossL2Prover contract
+     * @param _permit2 The address of the permit2 contract
+     * @param _localChainId The chain ID of the chain this contract is deployed on
+     */
+    constructor(
+        ICrossL2ProverV2 _prover,
+        address _permit2,
+        uint256 _localChainId
+    ) BasicSwap7683(_permit2) {
+        prover = _prover;
+        localChainId = _localChainId;
+    }
+
+    // ============ Admin Functions ============
+    function setDestinationContract(uint256 chainId, address contractAddress) external onlyOwner {
+        if (contractAddress == address(0)) revert InvalidDestinationContract();
+        destinationContracts[chainId] = contractAddress;
+        emit DestinationContractUpdated(chainId, contractAddress);
+    }
+
+    // ============ External Functions ============
+    /**
+     * @notice Process a settlement proof from a destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     * @param eventProof The proof of the Fill event from the destination chain
+     */
+    function handleSettlementWithProof(bytes calldata eventProof) external {
+        // Verify event using Polymer prover
+        (
+            uint32 provenChainId,
+            address emitter,
+            ,  // topics
+            bytes memory data
+        ) = prover.validateEvent(eventProof);
+
+        // Verify destination contract is registered for the proven chain
+        address expectedEmitter = destinationContracts[provenChainId];
+        if (expectedEmitter == address(0)) revert UnregisteredDestinationChain();
+
+        // Validate emitter matches registered destination
+        if (emitter != expectedEmitter) revert InvalidEmitter();
+
+        // Decode data from the Filled event format
+        (bytes32 eventOrderId, bytes memory originData, bytes memory fillerData) = abi.decode(data, (bytes32, bytes, bytes));
+
+        // Process settlement with the message origin, sender, order ID, and receiver
+        _handleSettleOrder(
+            provenChainId,
+            TypeCasts.addressToBytes32(emitter),
+            eventOrderId,
+            abi.decode(fillerData, (bytes32))
+        );
+
+        // _handleSettleOrder checks eligibility with
+        // _checkOrderEligibility and simply returns early without
+        // reverting if the order isn't eligible. we need to check if
+        // the order was settled successfully and revert if not.
+        if (orderStatus[eventOrderId] != SETTLED) revert SettlementFailed();
```

Comment:
It may be good idea to add a similar check into the `handleRefundWithProof` also

Reply:
for sure! added the same check